Tasks/DeployVisualStudioTestAgent/TestAgentConfiguration.ps1

```diff
@@ -615,6 +616,39 @@ function InvokeDTAExecHostExe([string] $Version, [System.Management.Automation.P
 
 }
 
+function CreateNewSession()
+{
+    $winrmconfigDetails = Winrm e  winrm/config/listener -format:pretty |Out-String
+    $xmldoc = [XML]$winrmconfigDetails    
+    $transports = $xmldoc.GetElementsByTagName("cfg:Transport")
+    $ports = $xmldoc.GetElementsByTagName("cfg:Port")
+   
+    if( $ports -ne $null -and  $transports -ne $null -and $ports.Count -gt 0)
+    {
+        $port = $ports[0].InnerText
+        $transport = $transports[0].InnerText
+    }
+    else
+    {
+        Write-Verbose -Message("Unable to fetch WinRM config details. Using default port for configuration") -Verbose
+        $port = 5985
+        $transport = HTTP          
+    }
+
+    Write-Verbose -Message("Using Port {0} for creating session" -f $port) -Verbose
+        
+    if( $transport -eq "HTTPS")
+    {
+        $sessionOption = New-PSSessionOption -SkipCACheck -SkipCNCheck
+        $session = New-PSSession -ComputerName . -Port $port -SessionOption $sessionOption -UseSSL
```

Comment:
you have removed credential usage from this ... any reason for that change ?
if no, then I guess its safer to stick with the old tested code